Amplify/Categories/DataStore/Model/Schema/AuthRule.swift

```diff
@@ -31,31 +24,25 @@ public typealias AuthRules = [AuthRule]
 public struct AuthRule {
     public let allow: AuthStrategy
     public let field: CodingKey?
-    public let provider: AuthProvider?
     public let ownerField: CodingKey?
     public let identityClaim: String?
     public let groupClaim: String?
     public let groups: [String]
-    public let groupsField: CodingKey?
```

Comment:
if the group field is customized by the developer isn't this needed? (genuine question, I'm not sure how it works)

Reply:
this is a valid question and i removed it because it was relating to "dynamic" groups, and thought we didn't really support this yet. but really what dynamic groups is is to specific the field on the model which is to contain the groups, just like how developer can override the ownerField to something other than "owner". 

Adding this back in to better have this info and follow up with testing expected functionality